cmd/swarm/upload.go

```diff
@@ -36,22 +36,67 @@ import (
 	"gopkg.in/urfave/cli.v1"
 )
 
+var (
+	MAX_STDIN_SIZE = 1024 * 1000 * 100
+)
+
+func swapStdin() (size int, path string) {
+	b := make([]byte, 1024)
+	size = 0
+
+	td, err := ioutil.TempDir("", "swarm-stdin")
+	if err != nil {
+		utils.Fatalf("Could not create temporary directory: %s", err)
+	}
+	path = filepath.Join(td, "data")
+	f, err := os.Create(path)
+	if err != nil {
+		utils.Fatalf("Could not create temporary file: %s", err)
+	}
+	r := io.TeeReader(os.Stdin, f)
```

Comment:
Rather than using a `TeeReader` and doing the bounds checking manually, you should use an [`io.LimitReader`](https://golang.org/pkg/io/#LimitedReader) (but I also wonder if there is a need for a limit?)